detection_rules/generic_loader.py

```diff
@@ -101,9 +102,9 @@ def _assert_new(self, item: GenericCollectionTypes) -> None:
         file_map = self.file_map
         name_map = self.name_map
 
-        assert not self.frozen, f"Unable to add item {item.name} {item.id} to a frozen collection"
+        assert not self.frozen, f"Unable to add item {item.name} to a frozen collection"
```

Comment:
For posterity sake: This is needed because kibana allows you to export rules with the same action id.

<details><summary>Example</summary>
<p>

```
{
    "id": "25bf2028-fbc0-49a4-b45c-474f62fc37e3",
    "updated_at": "2024-08-05T18:17:44.646Z",
    "updated_by": "3610252053",
    "created_at": "2024-08-04T19:02:29.366Z",
    "created_by": "eric.forte",
    "name": "test mail rule",
    "tags": [],
    "interval": "5m",
    "enabled": false,
    "revision": 3,
    "description": "here is a test file",
    "risk_score": 21,
    "severity": "low",
    "license": "",
    "output_index": "",
    "meta": {
        "from": "1m",
        "kibana_siem_app_url": "https://stryker8-12.kb.europe-west1.gcp.cloud.es.io:9243/app/security"
    },
    "author": [],
    "false_positives": [],
    "from": "now-360s",
    "rule_id": "a1a32864-24de-4314-94cd-39f859b56619",
    "max_signals": 100,
    "risk_score_mapping": [],
    "severity_mapping": [],
    "threat": [],
    "to": "now",
    "references": [],
    "version": 2,
    "exceptions_list": [],
    "immutable": false,
    "related_integrations": [],
    "required_fields": [],
    "setup": "",
    "type": "eql",
    "language": "eql",
    "index": [
        "logs-*"
    ],
    "query": "process where true\n",
    "filters": [],
    "actions": [
        {
            "group": "default",
            "id": "elastic-cloud-email",
            "params": {
                "message": "Rule {{context.rule.name}} generated {{state.signals_count}} alerts. my double space newline here  ### My header newline  More stuff here.",
                "subject": "test subject",
                "to": [
                    "test@all.org"
                ]
            },
            "action_type_id": ".email",
            "uuid": "cf7fc283-0463-4a21-a800-fab3550a7048",
            "frequency": {
                "summary": true,
                "notifyWhen": "onActiveAlert",
                "throttle": null
            }
        },
        {
            "group": "default",
            "id": "elastic-cloud-email",
            "params": {
                "message": "Rule {{context.rule.name}} generated {{state.signals_count}} alerts",
                "to": [
                    "eric.forte@elastic.co"
                ],
                "subject": "test"
            },
            "action_type_id": ".email",
            "uuid": "12c1e12c-7e9f-4959-872c-73d427494f70",
            "frequency": {
                "summary": true,
                "notifyWhen": "onActiveAlert",
                "throttle": null
            }
        }
    ]
}
{
    "id": "2e299dad-3a09-4c08-89cb-f08c4f85a18e",
    "updated_at": "2024-08-05T18:17:45.668Z",
    "updated_by": "3610252053",
    "created_at": "2024-08-04T19:02:30.366Z",
    "created_by": "eric.forte",
    "name": "TestActionRule",
    "tags": [],
    "interval": "5m",
    "enabled": true,
    "revision": 3,
    "description": "Test",
    "risk_score": 21,
    "severity": "low",
    "note": "None",
    "output_index": "",
    "meta": {
        "from": "1m",
        "kibana_siem_app_url": "https://dev-deployment-2c684a.kb.us-central1.gcp.cloud.es.io:9243/s/infosec/app/security"
    },
    "author": [
        "INFOSEC_TEST"
    ],
    "false_positives": [],
    "from": "now-360s",
    "rule_id": "e818bf1b-dcc8-4746-80fa-a155a94a7f6b",
    "max_signals": 100,
    "risk_score_mapping": [],
    "severity_mapping": [],
    "threat": [],
    "to": "now",
    "references": [],
    "version": 2,
    "exceptions_list": [],
    "immutable": false,
    "related_integrations": [],
    "required_fields": [],
    "setup": "",
    "type": "query",
    "language": "kuery",
    "index": [
        "apm-*-transaction*",
        "auditbeat-*",
        "endgame-*",
        "filebeat-*",
        "logs-*",
        "packetbeat-*",
        "traces-apm*",
        "winlogbeat-*",
        "-*elastic-cloud-logs-*"
    ],
    "query": "process.command_line : \"FakeRoot\"\n",
    "filters": [],
    "actions": [
        {
            "group": "default",
            "id": "elastic-cloud-email",
            "params": {
                "message": "Rule {{context.rule.name}} generated {{state.signals_count}} alerts",
                "subject": "TestEmail",
                "to": [
                    "eric.forte@elastic.co"
                ]
            },
            "action_type_id": ".email",
            "uuid": "c405d3e6-f47d-4ee2-bfc2-9fa786051c66",
            "frequency": {
                "summary": true,
                "notifyWhen": "onActiveAlert",
                "throttle": null
            }
        },
        {
            "group": "default",
            "id": "478b2165-83fb-480d-8a4a-bb47cfcafd4c",
            "params": {
                "body": "{}"
            },
            "action_type_id": ".webhook",
            "uuid": "fd2c83b6-414b-4e31-939a-27f399c06203",
            "frequency": {
                "summary": true,
                "notifyWhen": "onActiveAlert",
                "throttle": null
            }
        },
        {
            "group": "default",
            "id": "elastic-cloud-email",
            "params": {
                "message": "Rule {{context.rule.name}} generated {{state.signals_count}} alerts",
                "to": [
                    "eric.forte@elastic.co"
                ],
                "subject": "test"
            },
            "action_type_id": ".email",
            "uuid": "f3c769f0-148a-4c98-b401-ec9a675be856",
            "frequency": {
                "summary": true,
                "notifyWhen": "onActiveAlert",
                "throttle": null
            }
        }
    ]
}
{
    "id": "478b2165-83fb-480d-8a4a-bb47cfcafd4c",
    "type": "action",
    "updated_at": "2024-08-05T16:14:32.825Z",
    "created_at": "2024-08-05T16:14:32.825Z",
    "version": "WzU2NzcwNDAsMl0=",
    "attributes": {
        "actionTypeId": ".webhook",
        "isMissingSecrets": false,
        "name": "test",
        "config": {
            "hasAuth": false,
            "method": "post",
            "url": "https://dev-deployment-2c684a.kb.us-central1.gcp.cloud.es.io/"
        },
        "secrets": {}
    },
    "references": [],
    "managed": false,
    "coreMigrationVersion": "8.8.0",
    "typeMigrationVersion": "8.3.0"
}
{
    "exported_count": 3,
    "exported_rules_count": 2,
    "missing_rules": [],
    "missing_rules_count": 0,
    "exported_exception_list_count": 0,
    "exported_exception_list_item_count": 0,
    "missing_exception_list_item_count": 0,
    "missing_exception_list_items": [],
    "missing_exception_lists": [],
    "missing_exception_lists_count": 0,
    "exported_action_connector_count": 1,
    "missing_action_connection_count": 0,
    "missing_action_connections": [],
    "excluded_action_connection_count": 0,
    "excluded_action_connections": []
}
```

</p>
</details>